packages/react/src/vm-extension/node-vm-ext-list.tsx

```diff
@@ -0,0 +1,52 @@
+import { inject } from "@azure/bonito-core/lib/environment";
+import { NodeService } from "@batch/ui-service";
+import { BatchDependencyName } from "@batch/ui-service/lib/environment";
+import React from "react";
+import { VmExtensionList, VmExtItem } from "./vm-extension-list";
+
+interface NodeVmExtensionListProps {
+    accountEndpoint: string;
+    poolId: string;
+    nodeId: string;
+    onItemClick?: (item: VmExtItem) => void;
+}
+
+export const NodeVMExtList = (props: NodeVmExtensionListProps) => {
+    const { accountEndpoint, poolId, nodeId, onItemClick } = props;
+
+    const [extensions, setExtensions] = React.useState<VmExtItem[]>([]);
+    const [loading, setLoading] = React.useState<boolean>(true);
+
+    const nodeService: NodeService = React.useMemo(() => {
+        return inject(BatchDependencyName.NodeService);
+    }, []);
+
+    React.useEffect(() => {
+        setLoading(true);
+        nodeService
+            .listBatchNodeExtensions(accountEndpoint, poolId, nodeId)
+            .then((resList) => {
+                const exts = [];
```

Comment:
Need to check if the component is still mounted by the time this runs. To track this, you can return a callback which sets a local variable. See action-form.tsx for an example.

This is a pattern that's really hard to get right consistently. Probably a good case for a custom hook.

Reply:
To expand a bit on the issue, imagine the following hook:

```
useEffect(() => {
    loadSomeData()
        .then((data) => {
            setSomeState(data);
        });
}, [dependency1, dependency2])
```

Now imagine the hook is run multiple times in quick succession due to component rerendering. Every `loadSomeData()` promise might still be unresolved, and there's no guaranteed order in which they'll resolve. The callbacks will race, and the final data passed to `setSomeState()` will vary based on the order in which the callbacks fire.

Adding a check to see if the component is still mounted before setting state will resolve this:

```
useEffect(() => {
    let isMounted = true;
    loadSomeData()
        .then((data) => {
            if (isMounted) {
                setSomeState(data);
            }
        });
    return () => { isMounted = false; };
}, [dependency1, dependency2])
```

See also: https://react.dev/learn/synchronizing-with-effects#fetching-data